quinn-proto/src/connection.rs

```diff
@@ -1231,17 +1228,14 @@ impl Connection {
 
                         if self.side.is_client() {
                             // Client-only beceause server params were set from the client's Initial
-                            let params = self
-                                .tls
-                                .get_quic_transport_parameters()
-                                .ok_or_else(|| {
+                            let params = match self.tls.transport_parameters() {
+                                Ok(Some(params)) => Ok(params),
+                                Ok(None) => {
                                     debug!(self.log, "remote didn't send transport params");
-                                    ConnectionError::from(TransportError::PROTOCOL_VIOLATION)
-                                })
-                                .and_then(|x| {
-                                    TransportParameters::read(self.side, &mut io::Cursor::new(x))
-                                        .map_err(Into::into)
-                                })?;
+                                    Err(ConnectionError::from(TransportError::PROTOCOL_VIOLATION))
+                                }
+                                Err(e) => Err(e.into()),
```

Comment:
If we're using `?` below, we can get rid of the `into` and `from` calls above.

Reply:
Yup, nice.